pkg/api/common/const.go

```diff
@@ -371,3 +371,7 @@ const (
 	AntreaInstallCniChainCmd      = "install_cni_chaining"
 	AntreaNetworkPolicyOnlyMode   = "networkPolicyOnly"
 )
+
+// Taints
+// AADPodIdentityTaintKey is the node taint for AAD Pod Identity-enabled clusters before NMI daemonset is ready
+const AADPodIdentityTaintKey string = "node.kubernetes.io/aad-pod-identity-not-ready"
```

Comment:
An alternative to this approach: we could create a programmatically generated (at runtime during cluster provisioning) taint to ensure that workloads aren't able to bypass node scheduling order enforcement by including this well-known taint toleration.

Thoughts @mboersma @DavidParks8 @ritazh @Michael-Sinz @aramase ?

Reply:
I feel the random generation method is more vulnerable than a method which temporarily blocks IMDS because anyone who found the random label could bypass it by making a toleration on their deployment. It feels a bit like security through obscurity that gets weaker the longer a cluster is running. Since labels aren't treated like secrets, they may be exposed in logs as well.

Reply:
fair enough, it's only really viable during cluster creation, after that the taint is discoverable by scraping the k8s node api